Improve ClassPathManager performance, caching unfound entries

harrisric opened the pull request:
Fixes #306.

harrisric committed:
Improve ClassPathManager performance, caching unfound entries

Fixes #306 and adds a few small optimisations.

Co-authored-by: Alexander Kriegisch <Alexander@Kriegisch.name>
Signed-off-by: Alexander Kriegisch <Alexander@Kriegisch.name>

kriegaex commented:
@harrisric, you do not seem to have signed the [Eclipse Contributor Agreement (ECA)](https://www.eclipse.org/legal/ECA.php) yet. Please do that, otherwise I cannot accept your PR, even after verifying it technically and functionally. Thank you in advance.

kriegaex committed:
Add ClasspathManagerTestCase::testUnfoundClassPerformance

Performance regression test for #306, failing before the enhancement and
passing afterward.

Signed-off-by: Alexander Kriegisch <Alexander@Kriegisch.name>

kriegaex commented:
@harrisric, please note that I added a test in a new commit, then reordered the commits and force-pushed after slightly modifying your commit message. Does that look OK to you?

The ECA check now also passes, either due to my co-authorship or because you signed the ECA in the meantime.

kriegaex committed:
ClasspathManagerTestCase cosmetics

Signed-off-by: Alexander Kriegisch <Alexander@Kriegisch.name>

harrisric commented:
@kriegaex yes, that looks good to me, and yes I have signed the ECA since the PR was first added. Thanks!
